lib/core/Axios.js

```diff
@@ -45,20 +45,61 @@ Axios.prototype.request = function request(config) {
     config.method = 'get';
   }
 
-  // Hook up interceptors middleware
-  var chain = [dispatchRequest, undefined];
-  var promise = Promise.resolve(config);
-
+  // filter out skipped interceptors
+  var requestInterceptorChain = [];
+  var synchronousRequestInterceptors = true;
   this.interceptors.request.forEach(function unshiftRequestInterceptors(interceptor) {
-    chain.unshift(interceptor.fulfilled, interceptor.rejected);
+    if (typeof interceptor.runWhen === 'function' && interceptor.runWhen(config) === false) {
+      return;
+    }
```

Comment:
As `runWhen` is executed immediately, we need to try/catch of it.

Reply:
I'm a bit confused on this one. What do we want to do in the catch if the `runWhen` function fails? The function is provided by the consumer so we don't know what kind of errors it might throw. I would think we just want the program to fail at that point.

The axios configuration accepts other consumer provided functions such as `validateStatus` and `paramsSerializer`. Calls to those functions are not wrapped in try/catch blocks.